editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -3917,7 +3917,7 @@ void CanvasItemEditor::_notification(int p_what) {
 			EditorRunBar::get_singleton()->connect("stop_pressed", callable_mp(this, &CanvasItemEditor::_update_override_camera_button).bind(false));
 		} break;
 
-		case NOTIFICATION_PHYSICS_PROCESS: {
+		case NOTIFICATION_PROCESS: {
 			EditorNode::get_singleton()->get_scene_root()->set_snap_controls_to_pixels(GLOBAL_GET("gui/common/snap_controls_to_pixels"));
```

Comment:
It's a pre-existing issue, but re-setting this every frame (and thus re-querying the project settings for it) seems quite wasteful.

I would move it to the constructor, and connect to `ProjectSettings.settings_changed` to update.